test/test_examples.py

```diff
@@ -1091,6 +1095,8 @@ def test_causal_consistency(self):
 
         # Start Causal Consistency Example 2
         with client.start_session(causal_consistency=True) as s2:
+            assert s1.cluster_time is not None
+            assert s1.operation_time is not None
```

Comment:
For the same reason as above, let's move this asserts to outside the example (above "Start Causal Consistency Example 2"). Also I'm very confused how this assert helps at all since we're accessing `s1.cluster_time` again below. From earlier conversations I would have expected mypy to not narrow the type here since each property access can theoretically return a different instance.

Reply:
Fixed.  Yeah, that does seem like a limitation.